doc/subsystems/shell/shell.rst

```diff
@@ -435,4 +435,21 @@ the shell will only print the subcommands registered for this command:
 
 	  params  ping
 
+Shell as the logger backend
+***************************
+
+Shell instance can act as the :ref:`logger` backend. Shell ensures that log
+messages are correctly multiplexed with shell output. Log messages from logger
+thread are enqueued and processed in the shell thread. Logger thread will block
+for configurable amount of time if queue is full, blocking logger thread context
+for that time. Oldest log message is removed from the queue after timeout and
+new message is enqueued. Use the ``shell stats show`` command to retrieve
```

Comment:
I'd like to add that if user needs to check lost messages on their own, that's not ok. The fact that messages are being dropped should be actively shown to the user (as in #12062).